libflux/go/libflux/analyze.go

```diff
@@ -76,13 +76,31 @@ func Analyze(astPkg *ASTPkg) (*SemanticPkg, error) {
 	return p, nil
 }
 
+// Merge packages merges the files of a given input package into a given output package.
+// This function borrows the input and output packages, but does not own them. Memory
+// must still be freed by the caller of this function.
+func MergePackages(inPkg *ASTPkg, outPkg *ASTPkg) error {
```

Comment:
I think this function would better located in `parser.go`, since it's dealing with ASTs rather than semantic graphs.